ui/desktop/app/controllers/scopes/scope/projects/sessions/index.js

```diff
@@ -4,25 +4,58 @@
  */
 
 import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 import orderBy from 'lodash/orderBy';
+import { statusTypes } from 'api/models/session';
 
 export default class ScopesScopeProjectsSessionsIndexController extends Controller {
+  // =services
+
   // =attributes
 
+  queryParams = [{ targets: { type: 'array' } }];
+
+  @tracked targets = [];
+
+  // =methods
+
+  /**
+   * A list of sessions sorted by created time
+   * @type {[SessionModel]}
+   */
+  get sortedSessions() {
+    return orderBy(this.model.sessions, 'created_time', 'desc');
+  }
+
+  /**
+   * Returns targets that are associated will all sessions the user has access to
+   * @returns {[TargetModel]}
+   */
+  get availableTargets() {
+    const uniqueSessionTargetIds = new Set(
+      this.model.allSessions.map((session) => session.target_id),
+    );
+    return this.model.allTargets.filter((target) =>
+      uniqueSessionTargetIds.has(target.id),
+    );
+  }
+
+  /**
+   * Returns all status types for sessions
+   * @returns {[string]}
+   */
+  get sessionStatusTypes() {
+    return statusTypes;
+  }
+
   /**
-   * A list of sessions filtered to the current user and sorted by created time,
-   * and active/pending.
-   * @type {SessionModel[]}
+   * Sets the targets query param to value of selectedTargets
+   * to trigger a query and closes the dropdown
+   * @param {object} selectedTargets
    */
-  get sorted() {
-    const sessions = this.model;
-    // Sort sessions by created time descending (newest on top)
-    const sortedSessions = orderBy(sessions, 'created_time', 'desc');
-    return [
-      // then move active sessions to the top
-      ...sortedSessions.filter((session) => session.isAvailable),
-      // and all others to the end
-      ...sortedSessions.filter((session) => !session.isAvailable),
-    ];
```

Comment:
Do we still need this logic in the sorted sessions or is it not needed anymore?

Reply:
when initially refactoring this, my thoughts were that this logic was unnecessary since users can use the status filter. However, on second thought it doesn't hurt to have additional sorting for when filters are combined.

Reply:
I think I agree with you that we probably don't need it but we should just check in with design once they're back. I just wanted to check if this was an explicit decision

Reply:
I added ICU-12137 to check back in on this.